ipaserver/plugins/dogtag.py

```diff
@@ -274,6 +274,8 @@
 
 logger = logging.getLogger(__name__)
 
+pki_version = pki.util.Version(pki.specification_version())
```

Comment:
This would not work if the server is installed without the CA service. For instance:
- ipa server with CA installed on fedora 38, with pki 11.3.1
- ipa replica without CA installed on rawhide with the patch + pki 11.4.3
If a user perform ipa cert-revoke on the replica, the code that retrieves the version is executed on the replica and forces the use of uppercase, but the JSON call is directed to the server which supports only the lowercase.

Reply:
I can't reproduce this. For some reason my pki master build from Oct 24 is accepting both "reason" and "Reason" so not failing at all.

Reply:
I changed it to fetch the remote version and act on that instead. I still think this is a server-side problem since this is an API change.